feat: Astro 4.7 article (#1078)
withastro/astro.build: src/content/blog/astro-470.mdx

---
title: "Astro 4.7"
description: "Astro 4.7 is out now! This release includes major improvements to the API for making toolbar apps, a new way to keep yourself up to date, and more."
publishDate: "April 25, 2024"
authors:
  - erika
  - ema
  - matthew
  - nate

>   - bjorn

Reviewer: ```suggestion
  - bjorn
  - sarah
```

lol, I'm (half) joking, but I feel pretty invested and involved in these releases! :smile:

Author: There's an infinite amount of room for this, get in!

coverImage: "/src/content/blog/_images/astro-470/post-header-4.7.webp"
socialImage: "/src/content/blog/_images/astro-470/og-image-4.7.webp"
lang: "en"
---

import BlogSummary from "/src/components/BlogSummary.astro";
import updateCheckerImage from "/src/content/blog/_images/astro-470/update-checker.webp";
import BlogContentImage from "/src/components/BlogContentImage.astro"

<BlogSummary>
  Astro 4.7 is now available! This release includes major improvements to the API for making toolbar apps, more ways to keep your Astro project up to date, and more.
</BlogSummary>

Full release highlights include:

- [**Major improvements to the Dev Toolbar API**](#dev-toolbar-api-improvements)
- [**Update checker**](#update-checker)
- [**`allowJs: true` for strictest TypeScript preset**](#allowjs-true-for-strictest-typescript-preset)

To upgrade an existing project, use the automated `@astrojs/upgrade` CLI tool. Alternatively, upgrade manually by running the upgrade command for your package manager:

```sh
# Recommended:
npx @astrojs/upgrade

# Manual:
npm install astro@latest
pnpm upgrade astro --latest
yarn upgrade astro --latest
```

## Dev Toolbar API improvements

> Astro 4.7 includes major improvements to the API for making toolbar apps. One of our main goal this release was to make it easier to build and maintain toolbar apps, and we're excited to share these improvements with you!

Reviewer: ```suggestion
Astro 4.7 includes significant improvements to the API for making toolbar apps. One of our main goal this release was to make it easier to build and maintain toolbar apps, and we are excited to bring you:
```
Just trying not to repeat "improvements", so here's one way to do it!

> - `defineToolbarApp` helper that makes it easier to define toolbar apps, this is similar to the `defineConfig` helper for defining Astro config, or `defineMiddleware` for defining middleware.
> - `app` and `server` helpers to make it easier to interact with the toolbar and the server.
> - New starter project for building toolbar apps, to help you get started building your own toolbar apps in no time. Run `npm create astro@latest -- --template toolbar-app` and start building your toolbar app.
> - [Revamped documentation for building toolbar apps](https://docs.astro.build/en/reference/dev-toolbar-app-reference/).
> - [All new recipe guiding you through building a toolbar app from scratch](https://docs.astro.build/en/recipes/making-toolbar-apps/), including how to use JSX frameworks like React or Preact in your app.

Reviewer: ```suggestion
- a `defineToolbarApp()` helper that makes it easier to define toolbar apps. This pattern should feel familiar as it is similar to `defineConfig()` in your Astro config and `defineMiddleware()` for defining middleware.
- new `app` and `server` helpers to make it easier to send and receive messages between the toolbar and the server.
- a brand new starter project for building toolbar apps. This will help you get started building your own toolbar apps in no time. Run `npm create astro@latest -- --template toolbar-app` and start building your toolbar app.
- a [recipe guiding you step-by-step through building your own toolbar app](https://docs.astro.build/en/recipes/making-toolbar-apps/) from scratch, including how to build your app using JSX frameworks like React or Preact.
- revamped [documentation of the Dev Toolbar App API](https://docs.astro.build/en/reference/dev-toolbar-app-reference/), with all examples updated to show using the new methods.
```

We hope you'll enjoy these improvements and we can't wait to see what you build with them!

## Update checker

Starting from this release, Astro will now check for updates when you run the dev server. If a new version is available, you'll see a message in the terminal with instructions on how to update and in the dev toolbar.

> <BlogContentImage src={updateCheckerImage} alt="A screenshot of the message that Astro shows when an update is available." />

Reviewer: ```suggestion
<BlogContentImage src={updateCheckerImage} alt="A screenshot of the message that Astro shows in the terminal when an update is available: update > New version of Astro available: 4.6.3. Run npx @astrojs/upgrade to update." />
```

> In order to avoid spamming you with update messages, Astro will only check for updates once per 10 days and only if you're multiple versions behind. You can also disable this feature by running `astro preferences disable checkUpdates` or setting the `ASTRO_DISABLE_UPDATE_CHECK` environment variable to `false`.

Reviewer: ```suggestion
To avoid spamming you with too many update messages, Astro will only check for updates once every 10 days and only if you're multiple versions behind. You can disable this feature by running `astro preferences disable checkUpdates` or setting the `ASTRO_DISABLE_UPDATE_CHECK` environment variable to `false`.
```
Just clarifying whether or not this will be something configured in `astro.config.mjs`? If so, I'd show the config file here. It's a short release! We have the space! :smile:

Author: It is not! Only the preference and the env var.

## `allowJs: true` for strictest TypeScript preset

> Our `strictest` TypeScript preset previously included `allowJs: false`, to disable the usage of .js files completely. This proved to be a bit too strict for the average user and often caused confusion. We've now changed this to `allowJs: true`, which allows you to use .js files in your project.
> 
> This should not be a breaking change, but if you're using the `strictest` preset and would like to still disable .js files, you can set `allowJs: false` manually in your `tsconfig.json`.

Reviewer: ```suggestion
Our `strictest` TypeScript preset previously included `allowJs: false`, to disable the use of `.js` files completely. This proved to be a bit too strict for the average user and often caused confusion. We've now changed this to `allowJs: true`, which allows you to use `.js` files in your project.

This should not be a breaking change, but if you're using the `strictest` preset and would like to still disable `.js` files, set `allowJs: false` manually in your `tsconfig.json`.
```
Mostly the suggestion here is that we just tend to use code syntax for file extensions or capitals for the name (e.g. PDF or `.pdf`) So either JS or `.js` would be consistent with that. I would keep it `.js` myself because we are specifically talking about files and their extensions, but your choice!

---
title: "Astro 4.7"

> description: "Astro 4.7 is out now! This release includes major improvements to the API for making toolbar apps, a new way to keep yourself up to date, and more."
> publishDate: "April 25, 2024"
> authors:
>   - erika
>   - ema
>   - matthew
>   - nate
>   - bjorn
>   - sarah
> coverImage: "/src/content/blog/_images/astro-470/post-header-4.7.webp"
> socialImage: "/src/content/blog/_images/astro-470/og-image-4.7.webp"
> lang: "en"
> ---
> 
> import BlogSummary from "/src/components/BlogSummary.astro";
> import updateCheckerImage from "/src/content/blog/_images/astro-470/update-checker.webp";
> import BlogContentImage from "/src/components/BlogContentImage.astro"
> 
> <BlogSummary>
>   Astro 4.7 is now available! This release includes major improvements to the API for making toolbar apps, more ways to keep your Astro project up to date, and more.
> </BlogSummary>
> 
> Full release highlights include:
> 
> - [**Major improvements to the Dev Toolbar API**](#dev-toolbar-api-improvements)

Reviewer: ```suggestion
description: "Astro 4.7 is out now! This release includes astronomical improvements to the API for making toolbar apps, a new way to keep yourself up to date, and more."
publishDate: "April 25, 2024"
authors:
  - erika
  - ema
  - matthew
  - nate
  - bjorn
  - sarah
coverImage: "/src/content/blog/_images/astro-470/post-header-4.7.webp"
socialImage: "/src/content/blog/_images/astro-470/og-image-4.7.webp"
lang: "en"
---

import BlogSummary from "/src/components/BlogSummary.astro";
import updateCheckerImage from "/src/content/blog/_images/astro-470/update-checker.webp";
import BlogContentImage from "/src/components/BlogContentImage.astro"

<BlogSummary>
  Astro 4.7 is now available! This release includes *astronomical* improvements to the API for making toolbar apps, more ways to keep your Astro project up to date, and more.
</BlogSummary>

Full release highlights include:

- [**Out-of-this-world improvements to the Dev Toolbar API**](#dev-toolbar-api-improvements)
```
Was trying to think of an alternative to major, so here's one big chunk where the word occurs. Do you want to go with some combination of "astronomical/out-of-this-world"? 😂 

Down in the section itself, I used "significant", which could also be repeated. Words like "extensive" or "considerable" are other neutral choices. Your pick! Maybe you're even fine saying "major"!

Author: I'm a bit boring! I like significant. I think I'll use significant in the title, and considerable in the text.

- a brand new starter project for building toolbar apps. This will help you get started building your own toolbar apps in no time. Run `npm create astro@latest -- --template toolbar-app` and start building your toolbar app.
- a [recipe guiding you step-by-step through building your own toolbar app](https://docs.astro.build/en/recipes/making-toolbar-apps/) from scratch, including how to build your app using JSX frameworks like React or Preact.
- revamped [documentation of the Dev Toolbar App API](https://docs.astro.build/en/reference/dev-toolbar-app-reference/), with all examples updated to show using the new methods.

We hope you'll enjoy these improvements and we can't wait to see what you build with them!

## Update checker

Starting from this release, Astro will now check for updates when you run the dev server. If a new version is available, you'll see a message in the terminal with instructions on how to update and in the dev toolbar.

<BlogContentImage src={updateCheckerImage} alt="A screenshot of the message that Astro shows in the terminal when an update is available: update > New version of Astro available: 4.6.3. Run npx @astrojs/upgrade to update." />

> To avoid spamming you with too many update messages, Astro will only check for updates once every 10 days and only if you're multiple versions behind. You can disable this feature by running `astro preferences disable checkUpdates` or setting the `ASTRO_DISABLE_UPDATE_CHECK` environment variable to `false`.

Reviewer: ```suggestion
To avoid spamming you with too many update messages, Astro will only check for updates once every 10 days and will only show the notice if you're multiple versions behind. You can disable this feature by running `astro preferences disable checkUpdates` or setting the `ASTRO_DISABLE_UPDATE_CHECK` environment variable to `false`.
```
